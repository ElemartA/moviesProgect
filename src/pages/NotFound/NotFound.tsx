import React from "react";
import s from "./NotFound.module.scss";

class NotFound extends React.Component {
  render() {
    return (
      <div data-testid="notFoundPage" className={s.container}>
        <img src={"/404.png"} alt="страница не найдена" />
      </div>
    );
  }
}

export default NotFound;
