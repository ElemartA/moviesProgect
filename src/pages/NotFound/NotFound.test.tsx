import React from "react";
import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import NotFound from "./NotFound";

describe("NotFound", () => {
  it("renders MyForm component", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <NotFound />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
    screen.debug();
    expect(screen.getByRole("img")).toBeInTheDocument();
    expect(screen.queryByText(/Главная/i)).toBeNull();
  });
});
