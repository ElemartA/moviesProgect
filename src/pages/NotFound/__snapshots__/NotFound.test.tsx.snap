// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound renders MyForm component 1`] = `
<DocumentFragment>
  <div
    class="container"
    data-testid="notFoundPage"
  >
    <img
      alt="страница не найдена"
      src="/404.png"
    />
  </div>
</DocumentFragment>
`;
